lib/Reference/PublicationReferenceProvider.php

```diff
@@ -131,9 +131,15 @@ public function resolveReference(string $referenceText): ?IReference {
 				// TRANSLATORS By @$author in $subreddit_name_prefixed
 				$description = $this->l10n->t('By @%1$s in %2$s', [$postInfo['author'], $postInfo['subreddit_name_prefixed']]);
 				$reference->setDescription($description);
-				$thumbnailUrl = ($postInfo['thumbnail'] === 'self' || $postInfo['thumbnail'] === 'spoiler')
-					? $this->urlGenerator->linkToRouteAbsolute(Application::APP_ID . '.redditAPI.getAvatar', ['subreddit' => $subreddit])
-					: $this->urlGenerator->linkToRouteAbsolute(Application::APP_ID . '.redditAPI.getThumbnail', ['url' => $postInfo['thumbnail']]);
+				if ($postInfo['thumbnail'] == 'image') {
+					$thumbnailUrl = $this->urlGenerator->linkToRouteAbsolute(Application::APP_ID . '.redditAPI.getThumbnail', ['url' => $postInfo['url_overridden_by_dest']]);
+				}
+				else if ($postInfo['thumbnail'] === 'self' || $postInfo['thumbnail'] === 'spoiler') {
+					$thumbnailUrl = $this->urlGenerator->linkToRouteAbsolute(Application::APP_ID . '.redditAPI.getAvatar', ['subreddit' => $subreddit]);
+				}
+				else {
+					$thumbnailUrl = $this->urlGenerator->linkToRouteAbsolute(Application::APP_ID . '.redditAPI.getThumbnail', ['url' => $postInfo['thumbnail']]);
+				}
```

Comment:
```suggestion
				if ($postInfo['thumbnail'] === 'image') {
					$thumbnailUrl = $this->urlGenerator->linkToRouteAbsolute(Application::APP_ID . '.redditAPI.getThumbnail', ['url' => $postInfo['url_overridden_by_dest']]);
				} elseif ($postInfo['thumbnail'] === 'self' || $postInfo['thumbnail'] === 'spoiler') {
					$thumbnailUrl = $this->urlGenerator->linkToRouteAbsolute(Application::APP_ID . '.redditAPI.getAvatar', ['subreddit' => $subreddit]);
				} else {
					$thumbnailUrl = $this->urlGenerator->linkToRouteAbsolute(Application::APP_ID . '.redditAPI.getThumbnail', ['url' => $postInfo['thumbnail']]);
				}
```
Let's use the strict equality check (`===`) and use the code style used everywhere in NC (no new lines before else blocks, `else if` => `elseif`.